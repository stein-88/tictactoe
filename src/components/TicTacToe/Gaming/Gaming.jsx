import React from 'react'
import Btn from '@generics/Btn'
import { v4 as uuidv4 } from 'uuid'
import schema from './data'
import s from './Gaming.scss'

const Gaming = ({
 resetGame, currentPlayer, board, handleDispatch,
}) => {
	if (
		!schema
		|| !board
		|| !board.length
		|| !currentPlayer
		|| !handleDispatch
		|| !resetGame
	) return null

	const { turnof } = schema

	return (
		<section className="w-100">
			<section className="w-100 text-right pb-3">
				<Btn text="Reset" onClick={resetGame} />
			</section>
			<section className="w-100">
				{board.map((row, rowIndex) => (
					<div key={uuidv4()} className="col-lg-9 col-12 m-auto">
						<div className="row m-0">
							{row && row.length > 0
								&& row.map((colValue, colIndex) => (
									<div
										onClick={() => {
											if (colValue) return null
											return handleDispatch(rowIndex, colIndex)
										}}
										aria-hidden="true"
										data-testrefer={colValue ? s.noCursor : ''}
										key={uuidv4()}
										className={`col ${s.clickText} ${colValue ? s.noCursor : ''} ${s.tictok} ${s[`box${rowIndex}col${colIndex}`]}`}
									>
										{colValue || ''}
									</div>
								))}
						</div>
					</div>
				))}
			</section>
			<section className="w-100 text-right pt-3">
				<p className="h3">{`${turnof} ${currentPlayer}`}</p>
			</section>
		</section>
	)
}

export default Gaming
