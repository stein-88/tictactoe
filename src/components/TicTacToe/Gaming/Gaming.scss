.textPic {
  min-height: 455px;
}

.clickText {
  color: white;
  text-align: center;
  font-size: 80px;
  padding: 10px;
}

.tictok {
  height: 150px;
  width: 150px;
  float: left;
  border: 3px solid white;
  cursor: pointer;

  &.box0col0 {
    border-left: none;
    border-top: none;
  }

  &.box0col1 {
    border-top: none;
  }

  &.box0col2 {
    border-right: none;
    border-top: none;
  }

  &.box1col0 {
    border-left: none;
  }

  &.box1col1 {
    border: 3px solid white;
  }

  &.box1col2 {
    border-right: none;
  }

  &.box2col0 {
    border-left: none;
    border-bottom: none;
  }

  &.box2col1 {
    border-bottom: none;
  }

  &.box2col2 {
    border-right: none;
    border-bottom: none;
  }

  &.noCursor {
    cursor: default;
  }
}
