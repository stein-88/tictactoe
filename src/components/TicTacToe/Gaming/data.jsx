const schemagamingtictactoe = {
  turnof: 'Turn of the Player ',
};

export default schemagamingtictactoe;
