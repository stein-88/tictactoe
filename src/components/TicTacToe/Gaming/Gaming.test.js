import React from 'react'
import renderer from 'react-test-renderer'
import Gaming from './Gaming'

jest.mock('uuid', () => ({ v4: () => `uniquekey${Math.random(1000) * 100}` }))

const fullProps = {
	resetGame: () => null,
	currentPlayer: 'X',
	board: Array(3).fill(Array(3).fill(null)),
	handleDispatch: () => null
}
const { resetGame, currentPlayer, board, handleDispatch } = fullProps

describe('Test on Gaming Component', () => {
	test('Should not render without props', () => {
		const compGaming = renderer.create(<Gaming />)
		const jsonGaming = compGaming.toJSON()
		expect(jsonGaming).toBeNull()
	})
	test('Should not render with resetGame props only', () => {
		const compGaming = renderer.create(<Gaming resetGame={resetGame} />)
		const jsonGaming = compGaming.toJSON()
		expect(jsonGaming).toBeNull()
	})
	test('Should not render with currentPlayer props only', () => {
		const compGaming = renderer.create(<Gaming currentPlayer={currentPlayer} />)
		const jsonGaming = compGaming.toJSON()
		expect(jsonGaming).toBeNull()
	})
	test('Should not render with board props only', () => {
		const compGaming = renderer.create(<Gaming board={board} />)
		const jsonGaming = compGaming.toJSON()
		expect(jsonGaming).toBeNull()
	})
	test('Should not render with handleDispatch props only', () => {
		const compGaming = renderer.create(<Gaming handleDispatch={handleDispatch} />)
		const jsonGaming = compGaming.toJSON()
		expect(jsonGaming).toBeNull()
	})
	test('Should not render with 3 props only', () => {
		const compGaming = renderer.create(
			<Gaming
				handleDispatch={handleDispatch}
				board={board}
				currentPlayer={currentPlayer}
			/>
		)
		const jsonGaming = compGaming.toJSON()
		expect(jsonGaming).toBeNull()
	})
	test('Should render with full props', () => {
		const compGaming = renderer.create(<Gaming {...fullProps} />)
		const jsonGaming = compGaming.toJSON()
		expect(jsonGaming).toMatchSnapshot()
	})
	test('Should have 3 noCursor', () => {
		const mockFunc = jest.fn()
		const compGaming = renderer.create(
			<Gaming
				resetGame={resetGame}
				handleDispatch={handleDispatch}
				board={[
					[null, null, 'X'],
					[null, null, 'O'],
					['O', null, null]
				]}
				currentPlayer={currentPlayer}
			/>
		)
		const resultFinal = compGaming.root.findAllByProps({ ['data-testrefer']: 'noCursor' })
		const resultFim = compGaming.root.findAllByProps({ ['data-testrefer']: '' })
		resultFinal[0].props.onClick()
		resultFim[0].props.onClick()
		expect(mockFunc).not.toHaveBeenCalled()
		expect(resultFinal).toHaveLength(3)
	})
})