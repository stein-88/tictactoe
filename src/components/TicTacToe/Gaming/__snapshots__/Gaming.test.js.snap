// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on Gaming Component Should render with full props 1`] = `
<section
  className="w-100"
>
  <section
    className="w-100 text-right pb-3"
  >
    <span
      aria-hidden="true"
      className="btnstyle"
      onClick={[Function]}
    >
      Reset
    </span>
  </section>
  <section
    className="w-100"
  >
    <div
      className="col-lg-9 col-12 m-auto"
    >
      <div
        className="row m-0"
      >
        <div
          aria-hidden="true"
          className="col clickText  tictok box0col0"
          data-testrefer=""
          onClick={[Function]}
        >
          
        </div>
        <div
          aria-hidden="true"
          className="col clickText  tictok box0col1"
          data-testrefer=""
          onClick={[Function]}
        >
          
        </div>
        <div
          aria-hidden="true"
          className="col clickText  tictok box0col2"
          data-testrefer=""
          onClick={[Function]}
        >
          
        </div>
      </div>
    </div>
    <div
      className="col-lg-9 col-12 m-auto"
    >
      <div
        className="row m-0"
      >
        <div
          aria-hidden="true"
          className="col clickText  tictok box1col0"
          data-testrefer=""
          onClick={[Function]}
        >
          
        </div>
        <div
          aria-hidden="true"
          className="col clickText  tictok box1col1"
          data-testrefer=""
          onClick={[Function]}
        >
          
        </div>
        <div
          aria-hidden="true"
          className="col clickText  tictok box1col2"
          data-testrefer=""
          onClick={[Function]}
        >
          
        </div>
      </div>
    </div>
    <div
      className="col-lg-9 col-12 m-auto"
    >
      <div
        className="row m-0"
      >
        <div
          aria-hidden="true"
          className="col clickText  tictok box2col0"
          data-testrefer=""
          onClick={[Function]}
        >
          
        </div>
        <div
          aria-hidden="true"
          className="col clickText  tictok box2col1"
          data-testrefer=""
          onClick={[Function]}
        >
          
        </div>
        <div
          aria-hidden="true"
          className="col clickText  tictok box2col2"
          data-testrefer=""
          onClick={[Function]}
        >
          
        </div>
      </div>
    </div>
  </section>
  <section
    className="w-100 text-right pt-3"
  >
    <p
      className="h3"
    >
      Turn of the Player  X
    </p>
  </section>
</section>
`;
